nginx/Dockerfile

```diff
@@ -1,11 +1,17 @@
 # base build
-FROM nginx:1.19.6-alpine as base_nginx
+
+FROM nginx:1.24.0-alpine as base_nginx
```

Comment:
Looking at the [Dockerfile](https://github.com/nginxinc/docker-nginx/blob/1a8d87b69760693a8e33cd8a9e0c2e5f0e8b0e3c/stable/alpine/Dockerfile) for this image, it seems to be installing `nginx` from packages provided by nginx.org, or building them from source code. You are then attempting to use those nginx binaries with a `nginx-mod-http-lua` package provided by Alpine.

Simply changing this base image to `alpine:3.17` should work: `apk` will install `nginx` from Alpine repos as a dependency for `nginx-mod-http-lua`:

```
$ docker run -ti alpine:3.17 sh
Unable to find image 'alpine:3.17' locally
3.17: Pulling from library/alpine
c41833b44d91: Already exists
Digest: sha256:124c7d2707904eea7431fffe91522a01e5a861a624ee31d03372cc1d138a3126
Status: Downloaded newer image for alpine:3.17
/ # apk add nginx-mod-http-lua
fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/main/aarch64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/community/aarch64/APKINDEX.tar.gz
(1/8) Installing pcre (8.45-r2)
(2/8) Installing nginx (1.22.1-r0)
Executing nginx-1.22.1-r0.pre-install
Executing nginx-1.22.1-r0.post-install
(3/8) Installing nginx-mod-devel-kit (1.22.1-r0)
(4/8) Installing libgcc (12.2.1_git20220924-r4)
(5/8) Installing luajit (2.1_p20210510-r3)
(6/8) Installing lua-resty-lrucache (0.13-r0)
(7/8) Installing lua-resty-core (0.1.24-r0)
(8/8) Installing nginx-mod-http-lua (1.22.1-r0)
Executing busybox-1.35.0-r29.trigger
OK: 11 MiB in 23 packages
/ #
```

You might also want to change the way you deploy nginx configs. Instead of overwriting the whole `nginx.conf`, you can drop your configs into `/etc/nginx/conf.d/` and `/etc/nginx/http.d/`. This will ensure that configs managed by Alpine packages (like getting additional modules enabled automatically via `/etc/nginx/modules/*` when they are installed) keep working as intended.